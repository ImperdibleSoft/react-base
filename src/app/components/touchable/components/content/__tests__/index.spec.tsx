import * as React from 'react';
import { shallow } from 'enzyme';

import Content from '..';

describe('Content', () => {
  it('should render a text', () => {
    const options = {
      label: 'Add',
    };
    const component = shallow(<Content options={options} />);
    expect(component).toMatchSnapshot();
  });

  it('should render a text and a text-icon', () => {
    const options = {
      icon: 'plus',
      label: 'Add',
      iconLast: true,
    };
    const component = shallow(<Content options={options} />);
    expect(component).toMatchSnapshot();
  });

  it('should render a text and an icon', () => {
    const options = {
      icon: <img src="plus" />,
      label: 'Add',
      iconLast: true,
    };
    const component = shallow(<Content options={options} />);
    expect(component).toMatchSnapshot();
  });

  it('should render a text-icon', () => {
    const options = {
      icon: 'plus',
    };
    const component = shallow(<Content options={options} />);
    expect(component).toMatchSnapshot();
  });

  it('should render a text-icon and a text', () => {
    const options = {
      icon: 'plus',
      label: 'Add',
      iconLast: false,
    };
    const component = shallow(<Content options={options} />);
    expect(component).toMatchSnapshot();
  });

  it('should render an icon', () => {
    const options = {
      icon: <img src="plus" />,
    };
    const component = shallow(<Content options={options} />);
    expect(component).toMatchSnapshot();
  });

  it('should render an icon and a text', () => {
    const options = {
      icon: <img src="plus" />,
      label: 'Add',
      iconLast: false,
    };
    const component = shallow(<Content options={options} />);
    expect(component).toMatchSnapshot();
  });
});
