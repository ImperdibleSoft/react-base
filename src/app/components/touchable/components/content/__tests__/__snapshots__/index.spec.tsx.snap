// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content should render a text 1`] = `
<Fragment>
  <Label>
    Add
  </Label>
</Fragment>
`;

exports[`Content should render a text and a text-icon 1`] = `
<Fragment>
  <Label>
    Add
  </Label>
  <Space />
  <i
    className="fa fa-plus"
  />
</Fragment>
`;

exports[`Content should render a text and an icon 1`] = `
<Fragment>
  <Label>
    Add
  </Label>
  <Space />
  <img
    src="plus"
  />
</Fragment>
`;

exports[`Content should render a text-icon 1`] = `
<Fragment>
  <i
    className="fa fa-plus"
  />
</Fragment>
`;

exports[`Content should render a text-icon and a text 1`] = `
<Fragment>
  <i
    className="fa fa-plus"
  />
  <Space />
  <Label>
    Add
  </Label>
</Fragment>
`;

exports[`Content should render an icon 1`] = `
<Fragment>
  <img
    src="plus"
  />
</Fragment>
`;

exports[`Content should render an icon and a text 1`] = `
<Fragment>
  <img
    src="plus"
  />
  <Space />
  <Label>
    Add
  </Label>
</Fragment>
`;
